### ISSUE TITLE
Build problems

### ISSUE THREAD

**Issue by norulz** [opengribs/XyGrib#70]
@nohal @did-g
Hi, I'm back from being away from my desk for a while and am digging back into XyGrib. Right now I'm unable to build in Ubuntu getting the following:
```
/usr/bin/ld: /usr/local/lib/libopenjp2.a(thread.c.o): undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
In Window I'm also having issues with cmake getting it to find libraries
i.e. zlib2 (I have no package - only the include and lib)
```
CMake Error at C:/Program Files (x86)/CMake/share/cmake-3.11/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find BZip2 (missing: BZIP2_LIBRARIES) (found version "1.0.6")
Call Stack (most recent call first):
  C:/Program Files (x86)/CMake/share/cmake-3.11/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files (x86)/CMake/share/cmake-3.11/Modules/FindBZip2.cmake:51 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:122 (find_package)
```
Help will be welcome

**Comment by did-g** [comment]
Hi,
on linux remove your  local version of libopenjpeg and use the system one
if new enough.
Trusty is using 2.1.3 and it's too old, decoder doesn't compile.

I had the same error with travis but I was able to link if  libopenjpeg is
configured qith (cf .travis.xml):
cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_CODEC=OFF -DOPJ_USE_THREAD=OFF

Either static lib is incompatible with thread or there's a missing
-lpthread, always using shared library for openjp2 could help too.
ie
https://github.com/opengribs/XyGrib/pull/58/commits/4796cd25cf4075e1b3b904453fb0903bee798932


Looks like window CMake is a WIP...

Regards
Didier

**Comment by norulz** [comment]
All working well in Ubuntu after creating a clean environment.

I got cmake to agree on building in Windows but still have debug issues and QtCreator is freezing with a debug build. I needed to copy library dll's to the run directory which never happened before when running via the IDE (with Qmake). Must be an issue related to path but I can't find what to change.

I retested the the cmakelists.txt in Linux after all the Windows mods and it is still working so I must have done something right.
